### PR TITLE
Improve error message on missing root links

### LINKED ISSUE #329: Improve error message on missing root links
This is a possible error that we can obtain in logs when building and testing komodo:

```
AssertionError: The roots in the link-tree is not matching the roots defined in link_roots in dict
Roots defined: {'stable', '2019.12', 'deprecated-py2', '2021.02-py36', '2021.11-py36', 'testing', '2021.08-py36', '2022.04-py36', '2021.04-py36', '2021.06-py36', '2022.03-py36', '2022.10-py38', '2022.06-py38', 'deprecated-py36', '2021.10-py36', '2021.01-py36', '2021.12-py36', '2022.08-py38', 'deprecated', 'unstable', 'bleeding', '2021.09-py36', '2022.05-py36', '2022.01-py36', '2021.07-py36', '2020.08-py27', '2022.02-py36', '2022.09-py38', '2021.05-py36', 'stable-py36', 'stable-py2', '2022.05-py38', '2021.03-py36'}
Roots expected: {'stable', '2019.12', 'deprecated-py2', '2021.02-py36', '2021.11-py36', 'testing', '2021.08-py36', '2022.12-py38', '2022.04-py36', '2021.04-py36', '2021.06-py36', '2022.03-py36', '2022.10-py38', '2022.06-py38', 'deprecated-py36', '2022.05-py38', '2021.10-py36', '2021.01-py36', '2021.12-py36', '2022.08-py38', 'deprecated', 'unstable', 'bleeding', '2021.09-py36', '2022.05-py36', '2022.01-py36', '2021.07-py36', '2020.08-py27', '2022.02-py36', '2021.05-py36', 'stable-py36', 'stable-py2', '2022.09-py38', '2021.03-py36'}
Build step 'Execute shell' marked build as failure
```

A computer is better than a human to read the keys of these two dicts and say more precisely what the mismatch is.

### PR DESCRIPTION
**Pull request by jonathan-eq**
Resolves #329

### CODE REVIEW

File: tests/test_sanity_check.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from komodo.symlink.sanity_check import assert_root_nodes
+
+
+def test_assert_root_nodes_error_message():
+    link_dict = {
+        "links": {
+            "stable": "2012.01",
```
Comment (xjules) [inline tests/test_sanity_check.py:9]: `stable` is there 2x.

Comment (jonathan-eq) [reply tests/test_sanity_check.py:9]: Sveiseblind...
Fixed!

File: komodo/symlink/sanity_check.py [hunk 2]
```diff
@@ -82,10 +82,11 @@ def assert_root_nodes(link_dict):
     inferred_roots = _get_root_nodes(link_dict)
     if set(input_roots) != inferred_roots:
         raise AssertionError(
-            "The roots in the link-tree is not matching "
+            "The roots in the link-tree are not matching "
```
Comment (kwinkunks) [inline komodo/symlink/sanity_check.py:85]: I think, "The roots in the link-tree do not match " sounds more grammatical.

File: komodo/symlink/sanity_check.py [hunk 3]
```diff
@@ -82,10 +82,11 @@ def assert_root_nodes(link_dict):
...
             + "the roots defined in link_roots in dict\n"
```
Comment (kwinkunks) [inline komodo/symlink/sanity_check.py:86]: The dict is called `root_links` I think? At least, that's what it's called in the symlink config JSON file in the `komodo-releases` repo.

So maybe, "the roots defined in the root_links dict" is better?

File: komodo/symlink/sanity_check.py (hunk 2 above)
Comment (jonathan-eq) [reply komodo/symlink/sanity_check.py:85]: Agreed!

File: komodo/symlink/sanity_check.py (hunk 3 above)
Comment (jonathan-eq) [reply komodo/symlink/sanity_check.py:86]: Great catch!

**Review by xjules — APPROVED** [review]
I think it looks good now! :rocket:

**Pull request merged**